Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+
+Devtools offers a family of `install_*()` functions to address some needs beyond the reach of `install.packages()` or to make existing capabilities easier to access:
+
+```{r}
+library(devtools)
+
+funs <- as.character(lsf.str("package:devtools"))
+grep("^install_.+", funs, value = TRUE)
+```
+
+`devtools::install_github()` is the flagship example of a sub-family of functions that can download a package from a remote location that is not CRAN and do whatever is necessary to install it. `install_github()` and friends are actually provided by the [remotes package](https://remotes.r-lib.org) and are re-exported by devtools. The rest of the devtools/remotes `install_*()` functions are aimed at making things that are technically possible with base tooling a bit easier or more explicit, such as `install_version()` which installs a specific version of a CRAN package.
+
+ Analogous to `.Rbuildignore`, described in section \@ref(rbuildignore), `.Rinstignore` lets you keep files present in a package bundle out of the installed package. However, in contrast to `.Rbuildignore`, this is rather obscure and rarely needed.
+
+### In-memory package
+
+*TODO: putting something here, so I can comment in my PR.*
+
+To use a package, you must load it into memory. To use it without providing the package name (e.g. `install()` instead of `devtools::install()`), you need to attach it to the search path. R loads packages automatically when you use them. `library()` (and the later discussed `require()`) load, then attach an installed package.
+
+```{r, eval = FALSE}
+# Automatically loads devtools
+devtools::install()
+    
+# Loads and _attaches_ devtools to the search path
+library(devtools)
+install()
+```
+
+The distinction between loading and attaching packages is not important when you're writing scripts, but it's very important when you're writing packages. You'll learn more about the difference and why it's important in [search path](#search-path).
+
+`library()` is not useful when you're developing a package because you have to install the package first. In future chapters you'll learn about `devtools::load_all()` and RStudio's "Build and reload", which allows you to skip install and load a source package directly into memory:
+
+```{r, echo = FALSE}
+knitr::include_graphics("diagrams/loading.png")
+```
+
+## What is a library? {#library}
+
+A library is simply a directory containing installed packages. You can have multiple libraries on your computer. In fact, almost every one has at least two: one for packages you’ve installed, and one for the packages that come with every R installation (like base, stats, etc). Normally, the directories with user-installed packages vary based on the version of R that you’re using. That's why it seems like you lose all of your packages when you reinstall R --- they're still on your hard drive, but R can't find them. 
```

Comment:
Needs a segue

Reply:
Yeah nothing from "What is a library?" on has been touched in this PR. Will tackle separately.